### PR TITLE
add HOW-IT-WORKS.md

### PR DESCRIPTION
**Pull request by msbaek**
add HOW-IT-WORKS.md in order to exmplain the flow

### CODE REVIEW

**Comment by antoniolg** [comment]
The explanation is in the article linked in the Readme.md. What extra info do you need?

File: HOW-IT-WORKS.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+How it Works
+==========
+![](login-classes.png)
+
+1. View(Activity, Fragment, ...) calls presenter methods whenever there're user interaction
+2. Presenter implementation calls the interactor(use case handler) to get results from business/domain layer
+3. Interactor implementation returns the results or just returns the controll to presenter implementation by calling listener methods
```
Comment (antoniolg) [inline HOW-IT-WORKS.md]: Typo: "controll"

**Comment by antoniolg** [comment]
Great, thanks a lot @msbaek . Please do a commit --amend and push -f to fix the typo, so that it's only one commit, and I'll merge this PR.

**Comment by msbaek** [comment]
fix type, amend and push -f... all done

**Comment by antoniolg** [comment]
Awesome! Thanks

**Comment by msbaek** [comment]
:+1:

**Pull request merged**